src/acceleration_ros_publisher.cpp

```diff
@@ -58,22 +58,39 @@ void AccelerationRosPublisher::run()
         //std::cout << "thread " << this_id << " running...\n";
         //RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "running: '%d'", this_id);
 
-        auto started = std::chrono::high_resolution_clock::now();
-        bool success = false;
-        m_message = m_dataProvider->getData(success);
-        auto done = std::chrono::high_resolution_clock::now();
+        int32_t operation_mode = m_node->get_parameter("operation_mode").get_parameter_value().get<int32_t>();
```

Comment:
in general, all variables are declared at the beginning of a function and then only assigned where needed